File: Examples/DotNET/CSharp/Document/AppendDocuments.cs
using System;
using System.Collections.Generic;
using Com.Aspose.Words.Api;
using Com.Aspose.Words.Model;
using Com.Aspose.Storage.Api;

namespace Document
{
    class AppendDocuments
    {
        public static void Run()
        {
            // ExStart:1
            WordsApi wordsApi = new WordsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            // Set destination file name
            String destFileName = "test_doc.docx";

            // Set source file name
            String sourceFileName = "test_uploadfile.docx";

            Com.Aspose.Words.Model.DocumentEntryList body = new Com.Aspose.Words.Model.DocumentEntryList();
            List<Com.Aspose.Words.Model.DocumentEntry> docEntries = new List<Com.Aspose.Words.Model.DocumentEntry>();

            Com.Aspose.Words.Model.DocumentEntry docEntry = new Com.Aspose.Words.Model.DocumentEntry();
            docEntry.Href = sourceFileName;
            docEntry.ImportFormatMode = "KeepSourceFormatting";

            docEntries.Add(docEntry);
            body.DocumentEntries = docEntries;
            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(sourceFileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + destFileName));

                // Upload destination file to aspose cloud storage
                storageApi.PutCreate(destFileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + sourceFileName));

                // Invoke Aspose.Words Cloud SDK API to append word document
                Com.Aspose.Words.Model.DocumentResponse apiResponse = wordsApi.PostAppendDocument(destFileName, null, null, null, body);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    Console.WriteLine("Documents have been appended successfully");
                    
                    Com.Aspose.Storage.Model.ResponseMessage storageRes = storageApi.GetDownload(destFileName, null, null);
                    System.Diagnostics.Debug.WriteLine("response:" + storageRes.ResponseStream);
                    System.IO.File.WriteAllBytes(Common.GetDataDir() + sourceFileName, storageRes.ResponseStream);
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Perl/Document/AppendDocuments.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;
use AsposeWordsCloud::WordsApi;
use AsposeWordsCloud::ApiClient;
use AsposeWordsCloud::Configuration;
use AsposeWordsCloud::Object::DocumentEntryList;
use AsposeWordsCloud::Object::DocumentEntry;

my $configFile = '../Config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposeWordsCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeWordsCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeWordsCloud::Configuration::debug = 1;
$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Words API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $wordsApi = AsposeWordsCloud::WordsApi->new();

# Set input file name
my $name =  'SampleWordDocument.docx';
my $document1 =  'SampleWordDocument.docx';
my $document2 =  'SampleWordDocument.docx';

my @docEntry1 = AsposeWordsCloud::Object::DocumentEntry->new('Href' => $document1, 'ImportFormatMode' => 'KeepSourceFormatting');
my @docEntry2 = AsposeWordsCloud::Object::DocumentEntry->new('Href' => $document2, 'ImportFormatMode' => 'KeepSourceFormatting');
my $documentEntryList = AsposeWordsCloud::Object::DocumentEntryList->new('DocumentEntries' => [@docEntry1, @docEntry2]);

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);
# Invoke Aspose.Words Cloud SDK API to append word document   
$response = $wordsApi->PostAppendDocument(name=> $name, body=> $documentEntryList);

if($response->{'Status'} eq 'OK'){
	print "\nDocuments have been appended successfully";
	# Download appended document from storage server 
	my $output_file = $out_path. 'appended-'.$name;
	$response = $storageApi->GetDownload(Path => $name);;
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1